optimum/onnxruntime/trainer.py

```diff
@@ -1460,18 +1473,36 @@ def _export(
                 self.model.to("cpu")
             model = unwrap_model(self.model)
 
+        # TODO: Remove once `ORTModelForCausalLM` supports `loss` as an output
+        if "-with-past" in self.feature:
+            correct_feature = self.feature.replace("-with-past", "")
+            raise NotImplementedError(
+                "`use_cache` is not yet supported for ONNX Runtime inference in `ORTTrainer`, please replace "
+                f"{self.feature} task by {correct_feature}."
+            )
+
         onnx_config_constructor = TasksManager.get_exporter_config_constructor(
             model=model, exporter="onnx", task=self.feature
         )
         onnx_config = onnx_config_constructor(model.config)
-
         opset = onnx_config.DEFAULT_ONNX_OPSET if opset is None else opset
 
+        is_decoder = isinstance(onnx_config, OnnxConfigWithPast)
+
         if with_loss:
             onnx_config = wrap_onnx_config_for_loss(onnx_config)
             opset = max(opset, 12)  # Operators like `nll_loss`are added for opset>=12
 
-        _ = export(model, onnx_config, model_path, opset, device=device)
+        output_path = model_path / ONNX_DECODER_NAME if is_decoder else model_path / ONNX_WEIGHTS_NAME
+        _ = export(
+            model=model,
+            config=onnx_config,
+            opset=opset,
+            output=output_path,
+            device=device,
+        )
+
+        model.config.save_pretrained(model_path)
```

Comment:
Out of curiosity, what happens for encoder/decoder models? Do we save as monolith?

Reply:
This is the fix for `ORTTrainer`, the fix for `ORTSeq2SeqTrainer` is planned in another PR.

Reply:
I see thanks